src/apps/EDGE3D.hpp

```diff
@@ -0,0 +1,527 @@
+          /* Transform is:  Curl(w_i) <- (1/|J|)J^{T} Curl(w_i) */\
+          tdbasisx[m] = jxx*dbasisx[m] + jyx*dbasisy[m] + jzx*dbasisz[m];\
+          tdbasisy[m] = jxy*dbasisx[m] + jyy*dbasisy[m] + jzy*dbasisz[m];\
+          tdbasisz[m] = jxz*dbasisx[m] + jyz*dbasisy[m] + jzz*dbasisz[m];\
+          tdbasisx[m] *= inv_detj_pre;\
+          tdbasisy[m] *= inv_detj_pre;\
+          tdbasisz[m] *= inv_detj_pre;\
+        }\
+        /* Compute the local mass plus stiffness matrix */\
+        for (int m = 0; m < NB; m++) {\
+          const Real_type txm = tbasisx[m];\
+          const Real_type tym = tbasisy[m];\
+          const Real_type tzm = tbasisz[m];\
+          const Real_type dtxm = tdbasisx[m];\
+          const Real_type dtym = tdbasisy[m];\
+          const Real_type dtzm = tdbasisz[m];\
+          /* Compute the upper triangular portion */\
+          for (int p = m; p < NB; p++) {\
+            /* inner product: <w_i, w_j> */\
+            const Real_type txp = tbasisx[p];\
+            const Real_type typ = tbasisy[p];\
+            const Real_type tzp = tbasisz[p];\
+            const Real_type Mtemp = detjwgts*(txm*txp + tym*typ + tzm*tzp);\
+            /* inner product: <Curl(w_i), Curl(w_j)> */\
+            const Real_type dtxp = tdbasisx[p];\
+            const Real_type dtyp = tdbasisy[p];\
+            const Real_type dtzp = tdbasisz[p];\
+            const Real_type Stemp = detjwgts*(dtxm*dtxp + dtym*dtyp + dtzm*dtzp);\
+            /* Add the entries to the matrix*/\
+            const Real_type x = Mtemp + Stemp;\
+            edge_matrix[p*NB + m] = x;\
+            edge_matrix[m*NB + p] = x;\
+          }\
+        }\
+      }\
+    }\
+  }\
+  sum[iz] = 0.0;\
+  for (int m = 0; m < NB; m++) {\
+    Real_type check = 0.0;\
+    for (int p = 0; p < NB; p++) {\
+      check += edge_matrix[m*NB + p];\
+    }\
+    sum[iz] += check;\
+  }
+
+#include "common/KernelBase.hpp"
+
+namespace rajaperf
+{
+class RunParams;
+
+namespace apps
+{
+class ADomain;
+
+class EDGE3D : public KernelBase
+{
+public:
+
+  EDGE3D(const RunParams& params);
+
+  ~EDGE3D();
+
+  void setUp(VariantID vid, size_t tune_idx);
+  void updateChecksum(VariantID vid, size_t tune_idx);
+  void tearDown(VariantID vid, size_t tune_idx);
+
+  void runSeqVariant(VariantID vid, size_t tune_idx);
+  void runOpenMPVariant(VariantID vid, size_t tune_idx);
+  void runCudaVariant(VariantID vid, size_t tune_idx);
+  void runHipVariant(VariantID vid, size_t tune_idx);
+  void runOpenMPTargetVariant(VariantID vid, size_t tune_idx);
+
+  void setCudaTuningDefinitions(VariantID vid);
+  void setHipTuningDefinitions(VariantID vid);
+  template < size_t block_size >
+  void runCudaVariantImpl(VariantID vid);
+  template < size_t block_size >
+  void runHipVariantImpl(VariantID vid);
+
+private:
+  //static const size_t default_gpu_block_size = 256;
+  //using gpu_block_sizes_type = gpu_block_size::make_list_type<default_gpu_block_size>;
```

Comment:
Perhaps delete if not used?

Reply:
Any idea what this should be? I am inclined to uncomment this default value and delete what I used.

Reply:
Is this resolved?